pkg/gimletd/notifications/podStateManager.go

```diff
@@ -0,0 +1,131 @@
+package notifications
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/gimlet-io/gimlet-cli/pkg/dashboard/api"
+	"github.com/gimlet-io/gimlet-cli/pkg/dashboard/model"
+	"github.com/gimlet-io/gimlet-cli/pkg/dashboard/store"
+	"github.com/sirupsen/logrus"
+)
+
+type PodStateManager struct {
+	notifManager Manager
+	store        store.Store
+	waitTime     time.Duration
+}
+
+func NewPodStateManager(notifManager Manager, store store.Store, waitTime time.Duration) *PodStateManager {
+	return &PodStateManager{notifManager: notifManager, store: store, waitTime: waitTime}
+}
+
+func (p PodStateManager) Track(pods []*api.Pod) {
+	p.trackStates(pods)
+}
+
+func (p PodStateManager) Run() {
+	for {
+		p.setFiringState()
+
+		time.Sleep(p.waitTime * time.Minute)
+	}
+}
+
+func (p PodStateManager) trackStates(pods []*api.Pod) {
+	for _, pod := range pods {
+		podName := fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
+		currentTime := time.Now().Unix()
+
+		if p.alreadyAlerted(podName) || p.statusNotChanged(podName, pod.Status) {
+			continue
+		}
+
+		if podErrorState(pod.Status) {
+			err := p.store.SaveOrUpdatePod(&model.Pod{
+				Name:                podName,
+				Status:              pod.Status,
+				StatusDesc:          pod.StatusDescription,
+				AlertState:          "Pending",
+				AlertStateTimestamp: currentTime,
+			})
+			if err != nil {
+				logrus.Errorf("could't save or update pod: %s", err)
+			}
+		} else {
+			err := p.store.SaveOrUpdatePod(&model.Pod{
+				Name:                podName,
+				Status:              pod.Status,
+				StatusDesc:          pod.StatusDescription,
+				AlertState:          "OK",
+				AlertStateTimestamp: currentTime,
+			})
+			if err != nil {
+				logrus.Errorf("could't save or update pod: %s", err)
+			}
+		}
+	}
+}
+
+func (p PodStateManager) setFiringState() {
+	pods, err := p.store.Pods()
```

Comment:
This query resultset can grow very large:
- either make sure we remove deleted pods from the DB
- or limit the resultset with a query parameter. You only deal with pending pods (not sure?)
- or do both